### ISSUE TITLE
Zooming to points in inset not working correctly

### ISSUE THREAD

**Issue by hubertkozik** [highcharts/highcharts#17918]
#### Expected behaviour
If you want to make zoom to the point, which is the inset, it will zoom into that point in the inset.

#### Actual behaviour
If you zoom to a point, which is in the inset the zoom will be incorrectly calculated and will zoom into bad region of the map.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/6stLkbd8/
1) Click the 'Zoom to Texas' button - everything works correctly.
2) Click the 'Zoom to Alaska' button - incorrect behaviour, the point is in the inset.
2) Click the 'Zoom to Hawaii' button - incorrect behaviour, the point is in the inset.

#### Product version
Highmaps v10.3.1

**Comment by TorsteinHonsi** [comment]
@kamil-musialowski I got this working by translating the bounds first to pixels, then back to projected bounds on the _main_ map view (rather than the inset map view:

```ts
    public zoomTo(): void {
        const point = this as (MapPoint&MapPoint.CacheObject),
            chart = point.series.chart,
            mapView = chart.mapView;

        let bounds = point.bounds;

        if (mapView && bounds) {

            const inset = isNumber(point.insetIndex) &&
                mapView.insets[point.insetIndex];
            if (inset) {
                // If in an inset, translate the bounds to pixels ...
                const px1 = inset.projectedUnitsToPixels({
                        x: bounds.x1,
                        y: bounds.y1
                    }),
                    px2 = inset.projectedUnitsToPixels({
                        x: bounds.x2,
                        y: bounds.y2
                    }),
                    // ... then back to projected units in the main mapView
                    proj1 = mapView.pixelsToProjectedUnits({
                        x: px1.x,
                        y: px1.y
                    }),
                    proj2 = mapView.pixelsToProjectedUnits({
                        x: px2.x,
                        y: px2.y
                    });

                bounds = {
                    x1: proj1.x,
                    y1: proj1.y,
                    x2: proj2.x,
                    y2: proj2.y
                };

            }

            mapView.fitToBounds(bounds, void 0, false);

            point.series.isDirty = true;
            chart.redraw();
        }
    }
```

Before accepting it as it is, you should see if this (or a similar) operation is done elsewhere so we can avoid repeating it.

**Comment by TorsteinHonsi** [comment]
### Drop-in fix

Here's how to apply a drop-in fix to Highcharts v10.3.1: https://jsfiddle.net/highcharts/8tfzx6p4/
